internal/nodes/avi_model_evh_nodes.go

```diff
@@ -0,0 +1,1029 @@
+	if len(vsNode.VSVIPRefs) > 0 {
+		for i, fqdn := range vsNode.VSVIPRefs[0].FQDNs {
+			if utils.HasElem(hosts, fqdn) {
+				// remove logic conainer-lib candidate
+				vsNode.VSVIPRefs[0].FQDNs[i] = vsNode.VSVIPRefs[0].FQDNs[len(vsNode.VSVIPRefs[0].FQDNs)-1]
+				vsNode.VSVIPRefs[0].FQDNs[len(vsNode.VSVIPRefs[0].FQDNs)-1] = ""
+				vsNode.VSVIPRefs[0].FQDNs = vsNode.VSVIPRefs[0].FQDNs[:len(vsNode.VSVIPRefs[0].FQDNs)-1]
+			}
+		}
+	}
+}
+
+// RouteIngrDeletePoolsByHostname : Based on DeletePoolsByHostname, delete pools and policies that are no longer required
+func RouteIngrDeletePoolsByHostnameForEvh(routeIgrObj RouteIngressModel, namespace, objname, key string, fullsync bool, sharedQueue *utils.WorkerQueue) {
+	ok, hostMap := routeIgrObj.GetSvcLister().IngressMappings(namespace).GetRouteIngToHost(objname)
+	if !ok {
+		utils.AviLog.Warnf("key: %s, msg: nothing to delete for route: %s", key, objname)
+		return
+	}
+
+	utils.AviLog.Debugf("key: %s, msg: hosts to delete are :%s", key, utils.Stringify(hostMap))
+	for host, hostData := range hostMap {
+		shardVsName := DeriveHostNameShardVSForEvh(host, key)
+		if hostData.SecurePolicy == lib.PolicyPass {
+			shardVsName = lib.GetPassthroughShardVSName(host, key)
+		}
+		if shardVsName == "" {
+			// If we aren't able to derive the ShardVS name, we should return
+			utils.AviLog.Infof("key: %s, shard vs ndoe not found for host: %s", host)
+			return
+		}
+		modelName := lib.GetModelName(lib.GetTenant(), shardVsName)
+		found, aviModel := objects.SharedAviGraphLister().Get(modelName)
+		if !found || aviModel == nil {
+			utils.AviLog.Warnf("key: %s, msg: model not found during delete: %s", key, modelName)
+			continue
+		}
+		// Delete the pool corresponding to this host
+		if hostData.SecurePolicy == lib.PolicyEdgeTerm {
+			aviModel.(*AviObjectGraph).DeleteVsForHostnameForEvh(shardVsName, host, routeIgrObj, hostData.PathSvc, key, true, true, true)
+		} else if hostData.SecurePolicy == lib.PolicyPass {
+			aviModel.(*AviObjectGraph).DeleteObjectsForPassthroughHost(shardVsName, host, routeIgrObj, hostData.PathSvc, key, true, true, true)
+		}
+		if hostData.InsecurePolicy == lib.PolicyAllow {
+			aviModel.(*AviObjectGraph).DeleteVsForHostnameForEvh(shardVsName, host, routeIgrObj, hostData.PathSvc, key, true, true, false)
+		}
+		ok := saveAviModel(modelName, aviModel.(*AviObjectGraph), key)
+		if ok && len(aviModel.(*AviObjectGraph).GetOrderedNodes()) != 0 && !fullsync {
+			PublishKeyToRestLayer(modelName, key, sharedQueue)
+		}
+	}
+	// Now remove the secret relationship
+	routeIgrObj.GetSvcLister().IngressMappings(namespace).RemoveIngressSecretMappings(objname)
+	utils.AviLog.Infof("key: %s, removed ingress mapping for: %s", key, objname)
+
+	// Remove the hosts mapping for this ingress
+	routeIgrObj.GetSvcLister().IngressMappings(namespace).DeleteIngToHostMapping(objname)
+
+	// remove hostpath mappings
+	updateHostPathCacheV2(namespace, objname, hostMap, nil)
```

Comment:
You want to all these methods V2 or EVH?

Reply:
Only the methods which act on EvhVS struct and involve change in model ( VS, Pool, PG, Cert ). I have put it in this file, rest of the code is reused.